client/src/components/Nav/SettingsTabs/General.tsx

```diff
@@ -66,10 +66,39 @@ export const ClearChatsButton = ({
   );
 };
 
+export const LangSelector = ({
+  langcode,
+  onChange,
+}: {
+  langcode: string;
+  onChange: (value: string) => void;
+}) => {
+  const lang = useRecoilValue(store.lang);
+
+  return (
+    <div className="flex items-center justify-between">
+      <div>{localize(lang, 'com_nav_language')}</div>
+      <select
+        className="w-24 rounded border border-black/10 bg-transparent text-sm dark:border-white/20 dark:bg-gray-900"
+        onChange={(e) => onChange(e.target.value)}
+        value={langcode}
+      >
+        <option value="en">{localize(lang, 'com_nav_lang_english')}</option>
+        <option value="cn">{localize(lang, 'com_nav_lang_chinese')}</option>
+        <option value="it">{localize(lang, 'com_nav_lang_italian')}</option>
+        <option value="br">{localize(lang, 'com_nav_lang_brazilian_portuguese')}</option>
+        <option value="es">{localize(lang, 'com_nav_lang_spanish')}</option>
+      </select>
+    </div>
+  );
+};
+
 function General() {
   const { theme, setTheme } = useContext(ThemeContext);
   const clearConvosMutation = useClearConversationsMutation();
   const [confirmClear, setConfirmClear] = useState(false);
+  const langcode = useRecoilValue(store.lang);
+  const setLangcode = useSetRecoilState(store.lang);
```

Comment:
you can utilize useRecoilState instead of having two lines here for this. 

Example:   `const [conversation, setConversation] = useRecoilState(store.conversation);` from client\src\components\Input\AnthropicOptions\index.jsx

Reply:
Updated.